pkg/kube/structured/structured.go

```diff
@@ -235,6 +235,20 @@ func PersistentVolExists(kubeClientset kubernetes.Interface, name, expectedPhase
 	return nil
 }
 
+func PersistentVolClaimExists(kubeClientset kubernetes.Interface, name, expectedPhase string, namespace string) error {
+	vol, err := util.RetryOnError(&util.DefaultRetry, util.IsRetriable, func() (interface{}, error) {
+		return GetPersistentVolumeClaim(kubeClientset, name, namespace)
+	})
+	if err != nil {
+		return err
+	}
+	phase := string(vol.(*corev1.PersistentVolumeClaim).Status.Phase)
+	if phase != expectedPhase {
+		return fmt.Errorf("persistentvolumeclaim had unexpected phase %v, expected phase %v", phase, expectedPhase)
+	}
+	return nil
+}
```

Comment:
You are only retrying the call to GetPersistentVolumeClaim which I already has retry, you need to retry the whole thing:
``` go
PersistentVolClaimExists(){
  RetryOnError(){
    GetPersistentVolumeClaim(){
      RetryOnError(){ // already there
        GetPVC(){}
      }
    }
    CheckPhase(){}
  }
}
```

Reply:
yes, done.. thanks